test/e2e/specs/preview.test.js

```diff
@@ -6,96 +6,45 @@ import { parse } from 'url';
 /**
  * Internal dependencies
  */
-import '../support/bootstrap';
 import {
 	newPost,
-	newDesktopBrowserPage,
 	getUrl,
 	publishPost,
 } from '../support/utils';
 
 describe( 'Preview', () => {
-	beforeAll( async () => {
-		await newDesktopBrowserPage();
-	} );
-
 	beforeEach( async () => {
 		await newPost();
 	} );
 
-	let lastPreviewPage;
-
-	/**
-	 * Clicks the preview button and returns the generated preview window page,
-	 * either the newly created tab or the redirected existing target. This is
-	 * required because Chromium infuriatingly disregards same target name in
-	 * specific undetermined circumstances, else our efforts to reuse the same
-	 * popup have been fruitless and exhausting. It is worth exploring further,
-	 * perhaps considering factors such as origin of the interstitial page (the
-	 * about:blank placeholder screen), or whether the preview link default
-	 * behavior is used / prevented by the display of the popup window of the
-	 * same target name. Resolves only once the preview page has finished
-	 * loading.
-	 *
-	 * @return {Promise} Promise resolving with focused, loaded preview page.
-	 */
-	async function getOpenedPreviewPage() {
-		const eventHandlers = [];
-
-		page.click( '.editor-post-preview' );
-
-		const race = [
-			new Promise( ( resolve ) => {
-				async function onBrowserTabOpened( target ) {
-					const targetPage = await target.page();
-					resolve( targetPage );
-				}
-				browser.once( 'targetcreated', onBrowserTabOpened );
-				eventHandlers.push( [ browser, 'targetcreated', onBrowserTabOpened ] );
-			} ),
-		];
-
-		if ( lastPreviewPage ) {
-			race.push( new Promise( async ( resolve ) => {
-				function onLastPreviewPageLoaded() {
-					resolve( lastPreviewPage );
-				}
-
-				lastPreviewPage.once( 'load', onLastPreviewPageLoaded );
-				eventHandlers.push( [ lastPreviewPage, 'load', onLastPreviewPageLoaded ] );
-			} ) );
-		}
+	function sleep( time ) {
+		return new Promise( ( resolve ) => setTimeout( resolve, time ) );
+	}
```

Comment:
Puppeteer already provides this function in the form of [`page.waitFor`](https://github.com/GoogleChrome/puppeteer/blob/v1.4.0/docs/api.md#pagewaitforselectororfunctionortimeout-options-args).

Reply:
Ok, that's fine, can we add an inline comment with this explanation?

Reply:
Sure, in fact I've already changed this code locally and used the editor page to call `waitFor`, with an inline comment :)